### DISCUSSION TITLE
How was docsy converted to gsp / docToolchain?

### DISCUSSION THREAD

**Discussion by lprimak** [docToolchain/docToolchain#1085] [Q&A] [answered]
Let's say I am looking to use the latest docsy, or another template (Hugo, WordPress, Jekyll) or others with docToolchain.
How was docsy originally adapted to JBake and docToolchain? How can I adopt another non-gsp or non-freemarker templates to something that would work with docToolchain easily?

So, how was docsy originally adopted for docToolchain?
Do you have "magic sauce" or "recipe" how to get this accomplished and how this was accompished?
Thank you!

**Comment by rdmueller** [comment, marked as answer]
Here are some infos:

execute `./dtcw copyTheme` to copy the jBake theme to your project. This will overwrite the internal theme and you can try to create a new one.
When you finished your work, your can zip the site folder, put it somewhere on a server and reference it from dtcw as DTC_SITETHEME.

How did I adopt docs? I simply did a "save as" on the docsy page within the browser (strg+s). This gave me a plain HTML/CSS/JS version without any magic.

Then I took a look the structure of the original template and sliced the save page accordingly: Header, Footer, Main etc.

The biggest challenge is the menu because it is the most dynamic part. For this, you have to take a look at the original template.

Does this help?

**Reply by lprimak** [reply]
So basically manual effort correct?

thank you!

**Reply by rdmueller** [reply]
yes, but not too hard to do if you know about HTML and Groovy/Java

**Reply by mh182** [reply]
> Here are some infos:
> 
> execute `./dtcw copyTheme` to copy the jBake theme to your project. This will overwrite the internal theme and you can try to create a new one. When you finished your work, your can zip the site folder, put it somewhere on a server and reference it from dtcw as DTC_SITETHEME.

Looking at `dtcw` DTC_SITETHEME is not within the script. Meaning, it is not propagated into `DTC_OPTS`. Is the underlying Gradle project using environment variables apart from the ones provided on the call by `dtcw`?

**Reply by rdmueller** [reply]
@mh182 yes, `DTC_SITETHEM` is only mentioned in a comment

https://github.com/docToolchain/docToolchain/blob/ng/dtcw#L10

and used in `generateSite`

https://github.com/docToolchain/docToolchain/blob/ng/scripts/generateSite.gradle#L88

`DTC_HEADLESS` and `DTC_TEMPLATE0..9` are also variables which are used this way:

https://github.com/docToolchain/docToolchain/blob/ng/scripts/generateSite.gradle#L95

https://github.com/docToolchain/docToolchain/blob/ng/scripts/downloadTemplate.gradle#L17

**Reply by mh182** [reply]
Wow, I think we have to re-think the "provide configuration via environment"-architecture 

I think this discussion should definitely be part of the discussion in #1076.

Currently, those environment variables are "hardcoded" in `dtcw` so they are passed into the docker container. Whenever we add a new "module" with a new configuration option, `dtcw` would have to be adapted.

**Reply by mh182** [reply]
I created #1087 based on the discussion here.
